packages/perspective-jupyterlab/src/ts/psp_widget.ts

```diff
@@ -152,14 +153,20 @@ export class PerspectiveWidget extends Widget {
      * @param data
      */
     _update(data: TableData): void {
-        this.viewer.table.update(data);
+        if (this.viewer.table === undefined) {
+            this.viewer.load(perspective.worker().table(data));
+        } else {
+            this.viewer.table.update(data);
+        }
```

Comment:
As I mentioned in the #1296, I'd prefer not to replicate this behavior that was just removed from `<perspective-viewer>` itself.  I understand if we want to keep these actual methods on the `PerspectiveViewer` class so we do not need to model virtual `ipywidgets` interfaces for `Table` and `<perspective-viewer>`, but we can definitely remove any proxied code in Python which calls `update()` before `load()` on a `PerspectiveViewer`.

Reply:
One possible resolution is to move these specific changes to their own PR to capture the Issue, so we could merge this.

Reply:
I can also just create the table over [here](https://github.com/finos/perspective/blob/master/packages/perspective-jupyterlab/src/ts/renderer.ts#L50) and leave `PerspectiveWidget` alone. makes no difference to me, though someone should probably check that `client` only mode for the widget is still working

Reply:
Ah, so is this what prompted #1296?  If so yes, you should definitely construct the `Table` there.  Also - you should register this table as property of `PerspectiveDocumentWidget` and call its `delete()` method [here](https://github.com/finos/perspective/blob/master/packages/perspective-jupyterlab/src/ts/renderer.ts#L76) - or whatever the equivalent JupyterLab API is.

Reply:
Yep makes sense, will do it tomorrow